### ISSUE TITLE
MNIST sample - exeption by using 03_ConvBatchNorm.cntk

### ISSUE THREAD

**Issue by lenkele** [Microsoft/CNTK#405]
I tried sample with:
cntk configFile=../Config/03_ConvBatchNorm.cntk
And I'v got exeption: 
EXCEPTION occurred: Not yet implemented.

What I should do?

Thanks!

**Comment by Alexey-Kamenev** [comment]
Hello,
Are you running on GPU? Batch norm training is not yet implemented on CPUs though you can use models trained on GPU for inference on CPU.

**Comment by lenkele** [comment]
Yes, on CPU...
But if I run cntk configFile=../02_Convolution.cntk, I'v got equal probability vectors in the output on each test vector.  
All 10000 vectors in the output are the following:
 I'll be very grateful

**Comment by Alexey-Kamenev** [comment]
Just to make sure, if you don't use the write command and just use test, do you get expected accuracy (about 1%) on CPU?

**Comment by lenkele** [comment]
No!!
Final Results: Minibatch[1-625]: Samples Seen = 10000    err: ErrorPrediction/Sample = 0.9026    ce: CrossEntropyWithSoftmax/Sample = 2.405886    Perplexity = 11.08825 

This is exactly errror of the random hit into '9' by the highest value of the output vector...

**Comment by Alexey-Kamenev** [comment]
Are you using the latest version? This is what I get on my machine after 2 epochs on `02_Convolution.cntk` on CPU:
```
c:\src\cntk\x64\Release\CNTK.exe configFile=Config\02_Convolution.cntk RootDir=. deviceId=-1
...

Final Results: Minibatch[1-625]: SamplesSeen = 10000    err: ErrorPrediction/Sample = 0.0174    ce: CrossEntropyWithSoftmax/Sample = 0.053307069    Perplexity = 1.0547535

Action "test" complete.
```
Also note that what you get as a result of `write` command is not probabilities - those are the raw values of the output layer. See my response to #382

**Comment by lenkele** [comment]
Alexey, thank you for your answers.
I am using version from here: https://github.com/Microsoft/CNTK/wiki/CNTK-Binary-Download-and-Configuration
I took Binary again 3 days ago. 
I have windows 7 on my nodebook.
And I tried first:
cntk configFile=../Config/01_OneHidden.cntk

And it gives absolutely adequate results!
But this is too weak for my oun tasks (recognition of the symbols from foto), that's why I would like to try convolution.
Really I am looking forward to your advice. It is my first using of deep learning, I used boosting and something like this, I'm a mathematician and feature engineer in fact...

**Comment by lenkele** [comment]
I'v seen the newest version from 29.04. I take it and try cntk configFile=../02_Convolution.cntk.
 Now I have the new exeption:

EXCEPTION occurred: Undefined function or macro 'Pooling' in Pooling(inp, "max", {kW, kH, 1}, stride={hStride, vStride, 1}, autoPadding={true, true, false}, lowerPad=0, upperPad=0, imageLayout=cudnn)

**Comment by Alexey-Kamenev** [comment]
Hmm, the error that you are getting (`Undefined function or macro`) indicates that it's still the old version. Let me check our binaries once again.
Meanwhile, any chance you can compile the latest version from master sources?

**Comment by lenkele** [comment]
I think, I took last...
In this page: https://github.com/Microsoft/CNTK/releases
I took the "CNTK for Windows v.1.1 CPU only"
from the top of the page, from list:

Windows

CNTK for Windows v.1.1 CPU only
CNTK for Windows v.1.1 GPU
CNTK for Windows v.1.1 GPU with 1bit-SGD

**Comment by lenkele** [comment]
About compiling, I'm afraid that it is to complicated. I'm working in Visual studia 2013, but cuda and so on..

**Comment by Alexey-Kamenev** [comment]
Ok, let me check something on our side, will get back to you shortly...
Sorry for inconvenience.

**Comment by r-or** [comment]
Hi, I've got a similar problem, newest git master;

> /c/CNTK/x64/Release_CpuOnly/CNTK.exe configFile=Config/02_Convolution.cntk currentDirectory=Data deviceId=-1 command=test


yields
```
UCIFastReader: Starting at epoch 0, counting lines to determine record count...
 10000 records found.
starting epoch 0 at record count 0, and file position 0
already there from last epoch
RandomOrdering: 1989 retries for 10000 elements (19.9%) to ensure window condition
RandomOrdering: recached sequence for seed 0: 2334, 3830, ...
Final Results: Minibatch[1-625]: err = 0.89680000 * 10000; ce = 2.33810390 * 10000; perplexity = 10.36157132

Action "test" complete.
```

however I cannot check GPU version, because of lack of nvidia GPU...

**Comment by lenkele** [comment]
waiting for @Alexey-Kamenev responce...

**Comment by Alexey-Kamenev** [comment]
I cannot reproduce the `Undefined function or macro` error - I've tried both the latest binaries from our Releases page (CPU version and GPU version) as well as source code from master.

**Comment by Henry-White** [comment]
latest master (today)
[log.txt](https://github.com/Microsoft/CNTK/files/261174/log.txt)

CNTK.exe configFile=01_OneHidden.cntk 2>log.txt
EXCEPTION occurred: UCIParser::ParseInit - error opening file ../Data/Train-28x28.txt
log included.

**Comment by wolfma61** [comment]
The issue has been dormant for a long while. Please open a new issue if you still experience problems in the latest release.
